core-domain/src/main/kotlin/com/svenjacobs/app/leon/core/domain/sanitizer/salesforce/SalesforceParametersSanitizer.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Léon - The URL Cleaner
+ * Copyright (C) 2024 Sven Jacobs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.svenjacobs.app.leon.core.domain.sanitizer.salesforce
+
+import android.content.Context
+import com.svenjacobs.app.leon.core.common.regex.RegexFactory
+import com.svenjacobs.app.leon.core.domain.R
+import com.svenjacobs.app.leon.core.domain.sanitizer.RegexSanitizer
+import com.svenjacobs.app.leon.core.domain.sanitizer.Sanitizer
+import com.svenjacobs.app.leon.core.domain.sanitizer.SanitizerId
+
+class SalesforceParametersSanitizer :
+	RegexSanitizer(
+		// see https://help.salesforce.com/s/articleView?id=sf.mc_gai_utm_parameters.htm&type=5
+		RegexFactory.ofParameter(
+			"utm_source|utm_medium|utm_campaign|utm_term|utm_content|utm_id|" +
```

Comment:
> Under those circumstances, I feel the best solution is to add sfmc_ to the GoogleAnalyticsSanitizer and be done with it...?

I think it's absolutely okay to have a separate sanitizer for Salesforce because it's a different product.
Sanitizers can be enabled/disabled separately, so somebody might want to remove Google parameters but not Salesforce parameters 🤷🏼

Reply:
The way I understand it, the `utm_` stuff also is Salesforce  (see [their docs](https://help.salesforce.com/s/articleView?id=sf.mc_gai_utm_parameters.htm&type=5)), it's "just" that Google is also using/licensing it (apparently)...?

So, from a user control perspective, I feel that Salesforce should take care of both `utm_*` and `sfmc_*`. And possibly the Google sanitizer should also retain the removal of `utm_*` because that's what people are used to / where they attribute those parameters?

Reply:
> The way I understand it, the `utm_` stuff also is Salesforce (see [their docs](https://help.salesforce.com/s/articleView?id=sf.mc_gai_utm_parameters.htm&type=5)), it's "just" that Google is also using/licensing it (apparently)...?

`utm` stands for Urchin Tracking Module. [Urchin](https://en.wikipedia.org/wiki/Urchin_(software)) was an analytics software and company which was acquired by Google and incorporated into Google Analytics.

If Salesforce also uses `utm` parameters to delegate them to Google Analytics, I'm fine with this sanitizer also removing `utm_*`. But please use the mentioned `ofWildcardParameter` function.

```kotlin
RegexFactory.ofWildcardParameter("utm_|sfmc_")
```